### PR TITLE
[Android] Use auto theme everywhere, add option for auto dark in nav only

### PR DESCRIPTION
**Pull request by RedAuburn**
no reason to limit to just navigation IMO.

a user on telegram also complained about being flash-banged at night when cancelling routing, and said that it could be dangerous while driving.

### CODE REVIEW

**Comment by RedAuburn** [comment]
in fact, what about setting auto as the default setting?

**Comment by RedAuburn** [comment]
in testing, the theme isn't changed after opening when the theme should update. likely need to add a call in onstart or something

**Comment by RedAuburn** [comment]
now also updates theme in onResume(), not just app start & routing start.

https://github.com/user-attachments/assets/623b0031-5074-4246-8f47-aff26963bbdd

**Review by vng — APPROVED** [review]
@biodranik @rtsisyk 
Why it was not made like this (like in iOS) before? Do we have any objections?

File: android/app/src/main/java/app/organicmaps/util/ThemeSwitcher.java [hunk 1]
```diff
@@ -32,20 +32,17 @@ public void run()
...
-          theme = Config.getCurrentUiTheme(mContext);
-        }
-        else
-        {
-          boolean day = Framework.nativeIsDayTime(System.currentTimeMillis() / 1000,
-                                                  last.getLatitude(), last.getLongitude());
-          theme = (day ? defaultTheme : nightTheme);
-        }
+        theme = Config.getCurrentUiTheme(mContext);
+      }
```
Comment (AndrewShkrob) [inline android/app/src/main/java/app/organicmaps/util/ThemeSwitcher.java:40]: Nit: we don't use braces for one-liners:

```java
if (last == null)
  theme = Config.getCurrentUiTheme(mContext);
```

Comment (RedAuburn) [reply android/app/src/main/java/app/organicmaps/util/ThemeSwitcher.java:40]: done 👍️

**Review by AndrewShkrob — APPROVED** [review]

**Comment by RedAuburn** [comment]
Alex suggests adding a new option for auto in routing mode only, like https://github.com/organicmaps/organicmaps/issues/9161.
Will work on adding it

**Comment by RedAuburn** [comment]
have added a setting for only using auto dark mode when routing

**Comment by RedAuburn** [comment]
PTAL @vng @rtsisyk

**Comment by RedAuburn** [comment]
(nit fix)

**Review by vng — APPROVED** [review]
Did anyone test on a device?

I see in code that these 2 cases are always checked together. Is there any difference between them? Is there a place in code where only one of these settings affects?
```
ThemeUtils.isAutoTheme(context, res) || ThemeUtils.isNavAutoTheme(context, res)
```

**Comment by Misalf-git** [comment]
I'd like to suggest changing the string
`Auto` to `Automatic`.

At least in german it's rather confusing because
"Auto" can be short for "Automatisch",
but "Auto" is also the direct translation from "Car" (automobile).

Meaning
"On / Off / Auto" could be confused with
"On / Off / Car".

**Comment by vng** [comment]
Tested on a device - LGTM.

**Comment by pastk** [comment]
How is it supposed to work?

E.g. my phone is in the dark theme.

I switch to Night Mode ON, then to Auto - it just stays light.
I switch to Night Mode OFF, then to Auto - it just stays dark.
OM restarts doesn't help.

Is it because of the following?

> * auto theme needs a location fix to be calculated, it won't change without one (see [util/ThemeSwitcher.java#L38-L47](https://github.com/organicmaps/organicmaps/blob/a40ed508ec74882fa2611b376e099728b869d7f2/android/app/src/main/java/app/organicmaps/util/ThemeSwitcher.java#L38-L47))

**Comment by pastk** [comment]
Also how is it related to https://github.com/organicmaps/organicmaps/pull/8309 ? Is the latter needed still or could be closed?

**Comment by vng** [comment]
- Of course, the app should know your location for the Auto mode.
- https://github.com/organicmaps/organicmaps/pull/8309 can be rebased with adding FOLLOW_SYSTEM mode.

**Comment by pastk** [comment]
> * Of course, the app should know your location for the Auto mode.

IMO it should work with the last known location (likely we have to store it ourselves so that we don't rely on the system to supply it), otherwise the current behavior is pretty confusing..

**Comment by map-per** [comment]
> How is it supposed to work?

It's time based, so dark mode is enabled at night. But that's not clear form the name, we should change it to e.g. "At night" and "At night in navigation".

**Comment by pastk** [comment]
- https://github.com/organicmaps/organicmaps/issues/9729

**Comment by AD-Wright** [comment]
It looks like this has been implemented?  Awesome!! https://github.com/organicmaps/organicmaps/pull/8309

**Pull request merged**